xbmc/utils/DiscsUtils.cpp

```diff
@@ -0,0 +1,61 @@
+/*
+ *  Copyright (C) 2022- Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "DiscsUtils.h"
+
+#include "FileItem.h"
+//! @todo it's wrong to include videoplayer scoped files, refactor
+// dvd inputstream so they can be used by other components. Or just use libdvdnavdirectly
+#include "cores/VideoPlayer/DVDInputStreams/DVDInputStreamNavigator.h"
+
+#ifdef HAVE_LIBBLURAY
+//! @todo it's wrong to include vfs scoped files in an utils class, refactor
+// to use libbluray directly
+#include "filesystem/BlurayDirectory.h"
+#endif
+
+
+bool UTILS::DISCS::GetDiscInfo(UTILS::DISCS::DiscInfo& info, const std::string& mediaPath)
+{
+  // try to probe as a DVD
+  info = ProbeDVDDiscInfo(mediaPath);
+  if (!info.empty())
+    return true;
+  // try to probe as Blu-ray
+  info = ProbeBlurayDiscInfo(mediaPath);
+  return !info.empty();
```

Comment:
I have a couple suggestions, merely as a matter of style - what you've written here is fine and doesn't have to change, but I would add vertical whitespace, use sentence case, and structure the early returns to aid in future extensibility without dirtying git history (which also uses patterning to enhance readability):

```c++
// Try to probe as a DVD
info = ProbeDVDDiscInfo(mediaPath);
if (!info.empty())
  return true;

// Try to probe as Blu-ray
info = ProbeBlurayDiscInfo(mediaPath);
if (!info.empty())
  return true;

return false;
```